src/Extension/H264Rtp.cpp

```diff
@@ -203,8 +203,8 @@ void H264RtpEncoder::insertConfigFrame(uint64_t pts){
         return;
     }
     //gop缓存从sps开始，sps、pps后面还有时间戳相同的关键帧，所以mark bit为false
-    packRtp(_sps->data() + _sps->prefixSize(), _sps->size() - _sps->prefixSize(), pts, false, true);
-    packRtp(_pps->data() + _pps->prefixSize(), _pps->size() - _pps->prefixSize(), pts, false, false);
+    packRtp(_sps->data() + _sps->prefixSize(), _sps->size() - _sps->prefixSize(), pts, false, false);
+    packRtp(_pps->data() + _pps->prefixSize(), _pps->size() - _pps->prefixSize(), pts, false, true);
```

Comment:
Hi, thank for your patch, but this modification will lead to the mechanism of `gop cache` failed.
Decoder must first got `sps pps`, then will initialized successed.

Reply:
Sorry, didn't check wireshark


With these changes vlc works correctly

```patch
diff --git a/src/Extension/H264Rtp.cpp b/src/Extension/H264Rtp.cpp
index f12218a4..1024e534 100644
--- a/src/Extension/H264Rtp.cpp
+++ b/src/Extension/H264Rtp.cpp
@@ -203,8 +203,8 @@ void H264RtpEncoder::insertConfigFrame(uint64_t pts){
         return;
     }
     //gop缓存从sps开始，sps、pps后面还有时间戳相同的关键帧，所以mark bit为false
-    packRtp(_sps->data() + _sps->prefixSize(), _sps->size() - _sps->prefixSize(), pts, false, true);
-    packRtp(_pps->data() + _pps->prefixSize(), _pps->size() - _pps->prefixSize(), pts, false, false);
+    packRtpFu(_sps->data() + _sps->prefixSize(), _sps->size() - _sps->prefixSize(), pts, false, true);
+    packRtpFu(_pps->data() + _pps->prefixSize(), _pps->size() - _pps->prefixSize(), pts, false, false);
 }
 
 void H264RtpEncoder::packRtp(const char *ptr, size_t len, uint64_t pts, bool is_mark, bool gop_pos){
```

https://github.com/ZLMediaKit/ZLMediaKit/issues/2416#issue-1681232888
Please see this issue maybe there is another solution